src/Plugin/migrate_plus/data_parser/AuthenticatedXml.php

```diff
@@ -59,4 +59,17 @@ protected function openSourceUrl($url) {
 
   }
 
+  /**
```

Comment:
This change looks good to me. I haven't tested an actual source that has zero urls, however.

Reply:
It might be an issue with the way my source plugin works. I was migrating 6 basic images and would constantly see an exception in the Drupal logs between item 5 and 6 and it seemed that the migration was trying to get more URLs and there aren't any. It might also be due to the fact that for whatever reason the migrate_plus DataParserBase doesn't process the entire array of urls but rather [stops before the last one](https://cgit.drupalcode.org/migrate_plus/tree/src/DataParserPluginBase.php#n152).